backend/qemu.pm

```diff
@@ -874,6 +874,13 @@ sub start_qemu {
         for (my $i = 0; $i < $num_networks; $i++) {
             if ($vars->{NICTYPE} eq "user") {
                 my $nictype_user_options = $vars->{NICTYPE_USER_OPTIONS} ? ',' . $vars->{NICTYPE_USER_OPTIONS} : '';
+                if ($vars->{QEMU_ENABLE_SMBD}) {
+                    my $smbd_bin = find_bin('/usr/sbin/', 'smbd');
+                    $smbd_bin                             or die 'No server to provide SMB/CIFS services to SUT';
+                    (qx($smbd_bin --version) =~ /(\d.*)/) or warn 'smbd version regex was not matched';
+                    bmwqemu::diag "smbd version found on worker host: $1";
+                    $nictype_user_options .= ",smb=${\(dirname($basedir))}";
+                }
```

Comment:
indention gets a bit long. Could you please extract the code into a function and then simply call like

```perl
enable_qemu_smbd if $vars->{QEMU_ENABLE_SMBD};
```